### ISSUE TITLE
Submit expense by email

### ISSUE THREAD

**Issue by xdamman** [opencollective/opencollective#1862]
As the MVP, we want to focus on someone that takes a picture of a receipt and wants to submit it as an expense.

Today, they have to go to `opencollective.com/:collectiveSlug/expenses/new`, upload the photo and enter the details.

Members of the Open Collective Pro Tier should be able to send an email to `expenses@:collectiveSlug.opencollective.com`. This would create a `PENDING` expense with the photo in the email as the attachment.
We shouldn't try to do any OCR but the user should be able to specify the total amount in the subject or body.

It's beyond the scope of this issue, but if this can help designing the architecture, in a future version we would love to be able to forward a receipt received by email (without attachment). 
That means potentially developing a template architecture so that people can improve the code by adding support for different service providers. E.g. here is a way to detect that this is an email receipt from Lyft and this is how you can extract the amount.
People could forward or directly add the `expenses@:collectiveSlug.opencollective.com` email address as the billing email address so that they don't even have to forward anything.

**Comment by Betree** [comment]
These specs are good, having adapters based on the sender (with a smart default) is a very good idea to make this feature extensible. 

Having the `PENDING` status and requiring a manual validation is very important too because the `FROM` address in emails can be faked. 

Things that are not covered here:

- Do we prefill the payout method? How?

There's a possible answer in https://github.com/opencollective/opencollective/issues/1830#issuecomment-474579881. In short, we add the ability to add a set of rules in `/collective/expenses/automations` like if sender is ___@___.com, use this payout method and set the description to ______.

If no automation configured, set the payout field to null or manual.

- How to deal with emails with multiple attached documents

We can have a smart default here that loops through the attached files and takes the first one with `invoice` in it, falling back on the first pdf.

If it doesn't work for a given provider then we add a custom adapter.

**Comment by xdamman** [comment]
A first version could automatically create a `DRAFT` expense and send back a link that people can use to update it manually and confirm. Ideally, there could be a list somewhere of all the `DRAFT` expenses so that you can easily go through them all. 

Just that would already be a great feature. I could forward by email or configure the billing email to be `expenses@:collectiveSlug.opencollective.com` and once a week/month I just go through all the `DRAFT` expenses and confirm them. Already a big time saver on its own.

From there we could indeed think about improving the experience by defining default values based on recipes (or automation rules). 
E.g.
- if sender is billing@cloudflare.com set tags to `web service`, payout method to `paypal`, etc.
- if subject contains xxxxxx

If there are more than one attachment, I'd just pick the PDF if any, or the first image (with a threshold of minimum 100KB to avoid picking up inline images from signatures for example).

If there is no attachment, ideally we would want to take a screenshot of the email and use that as image. But that could be for later.

**Comment by Betree** [comment]
@xdamman What's the next step? Should we create a separate `opencollective-email-automations` service?

**Comment by Betree** [comment]
These kind of "integration" issues should, in my opinion, be implemented through Zapier.

Having a more powerful integration with Zapier that allows interactions with Open Collective's API would solve this in a very simple way, while allowing tons of other integrations for free. No need to create custom resolvers, email forwarding addresses...etc all we need to implement this is **the ability to create expenses through Zapier**. Then users just add a "Mail" trigger on Zapier, and use an action to create an expense with the attachment.

No need to add any code to configure a new provider. Works the same way if the original invoice is posted on a website or through a webhook.

However if we want to integrate more our API in Zapier we'll need to revamp the project, as what was made for #328 is not sustainable and we're already facing limitations.

**Comment by Betree** [comment]
Requested by @ninavizz in https://github.com/opencollective/opencollective/issues/4609#issue-984256814:

## User story 
As a paid contributor to an OpenCollective project, I would like a workflow for invoicing that aligns with the rest of my customers, so that invoicing my OC customers can take less time and be less annoying. 

Currently, most of my customers have a specific email address for their billing system's intake, that they send to their vendors plug-in to their invoicing software. Tagged fields in the email body can also map to filled-out form fields on today's form. 

TL;DR, it's a pain for me as a contributor to have to effectively do two separate workflows, to file an invoice for my one client who's an OpenCollective project. First step, on my invoicing software; and second, uploading it to my customer's OpenCollective profile on the website.

Nope, do not need any fancy callback features to tell my invoicing software (Harvest) you've paid me or anything.

## Best solution for this problem
A simple email address that projects can send to contributors, for us to plug-into our invoicing software as a "Send to:" customer email address. Bill.com does this, and its terrific.

## MVP
_What will be built as an MVP? (Please refer to our [wiki](https://github.com/opencollective/opencollective/wiki/Feature-Spec-Template) for what's acceptable as an MVP.)_
I am a freelancer.
My customer sends me an email address.
I plug that email address into my invoicing software so that when I "send" my invoice, it automagically uploads everything to the Open Collective form and I no longer have to endure that step on my own.

## Metrics 
?! I'm a UX contributor. I'm offering my feedback that this would be a beneficial feature that will save me time-on-task in my invoicing. Time on task for other people who submit invoices to OC, I guess.

## Documentation 
I'm not a developer. Look at how bill.com does it?

**Comment by Betree** [comment]
If we go back to this issue, we may consider using some third-party APIs out there (https://bill.com ?) that can provide us with structured info from PDF files as a service. If not too expensive, such a service would drastically reduce the complexity required to implement this feature. Parsing PDFs is a painful job.

**Comment by ninavizz** [comment]
I the spirit of "MVP" it could also be a simpler option to just create that email address that automagically posts a PDF to a project's "invoices;" and then the humans with the project that have to approve the expense, plug-in information themselves, manually. After the vendor has setup their "profile" in OC w/ payout options, etc.

Literally the only "new" information I ever post into the webform, is the total $$ and a subject for the expense.  I don't need to verify my address and payout info with each expense. 

Having project leads either cut-and-paste or compose "Expense" titles would also provide some uniformity for community members, when they review named expenses; eg, different contributors will interpret the ask for that information differently, and put different information in there... which can confuse donors (which I am guilty of!).

**Comment by xdamman** [comment]
This looks like a very good AI model to automatically parse receipts: https://github.com/clovaai/donut
It could be trained on the database of open collective for even better accuracy.

**Comment by Betree** [comment]
@xdamman Extracted this in https://github.com/opencollective/opencollective/issues/6793 as having some kind of automatic expense generation from images could be helpful for many different features (including this one).

**Comment by xdamman** [comment]
Not all hosts have virtual cards. So maybe, but I'd also make it available regardless.
It feels like already a v2 to link this to virtual cards and take care of reconciliations.
A first version should just handle the case of submitting an invoice/expense as a draft expense to be reviewed and approved (bonus if it can extra metadata using the chatgpt api, it's cheap enough).

**Comment by shannondwray** [comment]
Request for this from Homebrew - https://opencollective.com/homebrew 

They would love to be able to pop down a specific email for billing that automatically takes the expenses and files them on the platform
